LiteCore/Database/LiveQuerier.cc

```diff
@@ -79,7 +79,11 @@ namespace litecore {
 
     void LiveQuerier::stop() {
         logInfo("Stopping");
-        _stopping = true;
+         _backgroundDB->dataFile().useLocked([&](DataFile *df) {
+             // CBL-2335: Guard access to the _stopping variable so that
+             // it is not changed at unpredictable times
+             _stopping = true;
```

Comment:
Because atomicity does not protect from races.  We need to control when this variable is touched to avoid checking it, continuing onto a part that assumes it is false and then while that part is running modifying it.

Reply:
Question is, what's difference to move "_stopping = true" from before "useLocked" into "useLocke"?